R/parse.R

```diff
@@ -51,18 +51,24 @@ parse_delim <- function(x, delim, quote = "\"", ...) {
   )
 }
 
-parse_name_equals_value <- function(x) {
-  loc <- regexpr("=", x, fixed = TRUE)
-  pieces <- regmatches(x, loc, invert = TRUE)
+parse_name_equals_value <- function(x, safe = "&", quote = "\"") {
+  if (length(x) == 0) {
+    return(NULL)
+  }
+  # Note: quotes are removed in the parse_delim function
+  # re-introduce quotes to create a safe pattern to split on
```

Comment:
Why do we need quotes and `&` to create the safe pattern?

I think it would also be useful to add a comment pointing out that we're replacing the first instance of `=` with a sequence that we know doesn't exist in the string to ensure that we always produce at most two pieces.